File: examples/record/prepare/prepare_record.ts
import { RecordClient } from '@bloock/sdk';
import fs from 'fs';

try {
  const recordClient = new RecordClient();
  const record = await recordClient.fromString('Hello world').build();

  // we can get the hash of the record
  const hash = await record.getHash();
  console.log(hash);

  // build a record from an existing record
  const _record2 = await recordClient.fromRecord(record).build();

  // we can read a file as an array of bytes
  const file = fs.readFileSync('sample.pdf');
  const _record3 = await recordClient.fromFile(file).build();
} catch (e) {
  console.log(e);
}

File: examples/record/prepare_record/prepare_record.go
package main

import (
	"log"
	"os"

	"github.com/bloock/bloock-sdk-go/v2/client"
)

func main() {
	recordClient := client.NewRecordClient()
	record, err := recordClient.FromString("Hello world").Build()
	if err != nil {
		log.Println(err)
	}

	// we can get the hash of the record
	hash, err := record.GetHash()
	if err != nil {
		log.Println(err)
	}
	log.Println(hash)

	// build a record from an existing record
	_, err = recordClient.FromRecord(record).Build()
	if err != nil {
		log.Println(err)
	}

	// we can read a file as an array of bytes
	file, err := os.ReadFile("sample.pdf")
	if err != nil {
		log.Println(err)
	}

	// and build a record from it
	_, err = recordClient.FromFile(file).Build()
	if err != nil {
		log.Println(err)
	}
}
